assets/js/view/part/track_curvilinear_path.rb

```diff
@@ -32,38 +32,61 @@ def compute_track_rotation_degrees(edge_a, edge_b)
         end
       end
 
+      def load_from_tile
+        edge_a, edge_b = @path.exits
+
+        @curvilinear_type = compute_curvilinear_type(edge_a, edge_b)
+        @rotation = compute_track_rotation_degrees(edge_a, edge_b)
+      end
+
       def preferred_render_locations
+        regions =
+          case @curvilinear_type
+          when SHARP
+            [13, 14, 15, 21]
```

Comment:
maybe we can turn all these things into constants so they aren't reallocated every time

Reply:
40b75db